### PR TITLE
Switch to passing `—conf` for spark.executor

### PR DESCRIPTION
**Pull request by dragos**
Minor refactoring:

- added some docs
- increased driver memory (1.6 requires a minimum)

### CODE REVIEW

**Comment by skonto** [comment]
i was preparing the same PR for increasing memory :)

**Comment by dragos** [comment]
Actually, a nice outcome of this PR is that it will allow us to use `spark.mesos.executor.home`, skipping the (sometimes slow) copying of Spark tgz in each Mesos container sandbox.

Right now it won't work because Spark is using `getCanonicalPath` on this variable, which resolves it to an absolute path on the client machine (this is always wrong, going to open a ticket/PR)

**Comment by skonto** [comment]
Ok i will review it.

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/ClientModeRunner.scala [hunk 1]
```diff
@@ -32,13 +32,16 @@ object ClientModeRunner {
...
-        s"""--conf spark.executor.extraJavaOptions="$logFileConfig""""
-      )
+        s"""--conf spark.executor.extraJavaOptions="$logFileConfig"""")
 
       if (config.hasPath("spark.driver.host")) {
         sparkSubmitJobDesc += s"--conf spark.driver.host=${config.getString("spark.driver.host")}"
       }
 
+      if (config.hasPath("spark.executor.uri")) {
+        sparkSubmitJobDesc += s"""--conf spark.executor.uri="${config.getString("spark.executor.uri")}""""
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/ClientModeRunner.scala]: This is wrong, the internal quotes should not be there (I fixed it in the ClusterModeRunner but missed it here)

**Comment by skonto** [comment]
i verified its broken looks for a local path on the machine:

sh: 1: /tmp/mesos/slaves/77154d7c-8a2c-48d5-a50a-21729ca37102-S1/frameworks/77154d7c-8a2c-48d5-a50a-21729ca37102-0004/executors/driver-20151216122517-0001/runs/7fa508a5-4c06-4bd0-9b5b-4af8c7db47d0/spark-1.5.1-bin-hadoop2.6/bin/spark-class: not found

**Comment by dragos** [comment]
Yeah, 1.5.2 behaves differently. Note that the driver is launched, and the task starts fine, but the `spark-submit` call of the driver (inside the cluster) misses the `spark.executor.uri`):

```
Registered executor on 192.168.99.100
Starting task driver-20151216154450-0001
sh -c 'cd spark-1*;  bin/spark-submit --name com.typesafe.spark.test.mesos.framework.runners.SparkJobRunner --class com.typesafe.spark.test.mesos.framework.runners.SparkJobRunner --master mesos://192.168.99.100:5050 --driver-cores 1.0 --driver-memory 512M ../mesos-spark-integration-tests-assembly-0.1.0.jar http://192.168.99.100:5050 cluster spark_role spark:only 2 10.0.2.2 8888'
Forked command at 3298
Command exited with status 0 (pid: 3298)
```

**Comment by skonto** [comment]
Unless this is easily fixed i suggest we move mem increase in a different PR so we unblock jenkins along with SPARK_HOME fix in 1.6...

**Comment by dragos** [comment]
Yes, let's unblock mr. Jenkins. Will you open the PR?

**Comment by skonto** [comment]
:) yes its ready...

**Comment by dragos** [comment]
I removed the changes regarding `mesos.executor.home`, since they depend on fixing [SPARK-12423](https://issues.apache.org/jira/browse/SPARK-12423) in the first place. I'll open a separate PR for that one.

**Comment by skonto** [comment]
Cool i will review it :+1:

**Comment by skonto** [comment]
I checked out the pr as is and all tests in cluster mode fail with:

1.5.1:
sh: 1: /tmp/mesos/slaves/933c357b-6bae-4ad6-b27d-2e80a6430ada-S1/frameworks/933c357b-6bae-4ad6-b27d-2e80a6430ada-0004/executors/driver-20151220124750-0001/runs/72c527c3-43d5-4f64-b933-5828e82f4f5f/spark-1.5.1-bin-hadoop2.6/bin/spark-class: not found

1.5.2
sh: 1: /tmp/mesos/slaves/35b619d8-fa0c-4a84-89d1-8056345ed16b-S1/frameworks/35b619d8-fa0c-4a84-89d1-8056345ed16b-0004/executors/driver-20151220131041-0001/runs/61b80829-1c4e-4419-ba66-5d7ecbc87c09/spark-1.5.2-bin-hadoop2.6/bin/spark-class: not found

master:
sh: 1: /tmp/mesos/slaves/7135a7b9-b6b2-4a70-b658-d4e85a7c38b9-S1/frameworks/7135a7b9-b6b2-4a70-b658-d4e85a7c38b9-0004/executors/driver-20151220135527-0001/runs/c0e40b9c-2db5-4fa1-b978-a1b3a09b66d5/spark-2.0.0-SNAPSHOT-bin-test/bin/spark-class: not found
1.6.0-rc3 :
fails at submit level path is resolved to host...
bin/spark-submit: line 27: /home/stavros/workspace/installs/spark-master/spark-test/spark-1.6/spark/spark-1.6.0-bin-test/bin/spark-class: No such file or directory


This is the command for the submit:
Running command: /home/stavros/workspace/installs/spark-master/spark-test/spark/spark-2.0.0-SNAPSHOT-bin-test/bin/spark-submit --class com.typesafe.spark.test.mesos.framework.runners.SparkJobRunner --master mesos://thinkpad:7088 --driver-memory 512mb --deploy-mode cluster --conf spark.executor.uri=/var/spark/spark-2.0.0-SNAPSHOT-bin-test.tgz hdfs://172.17.0.1:8020/app/mesos-spark-integration-tests-assembly-0.1.0.jar http://172.17.0.1:5050 cluster spark_role spark:only 2 172.17.0.1 8888 with env vars: (MESOS_NATIVE_JAVA_LIBRARY,/usr/local/lib/libmesos.so)

All tests used same mesos library version 0.25.

**Comment by skonto** [comment]
Right now our tests in cluster mode pass in 1.5.2, master(2.0.0) and fail in 1.6.0.-rc3
check comments in #39 ...

**Comment by dragos** [comment]
Ah, I didn't see your comment, I will rebase and test again. All cluster mode tests worked for me, I have no idea what happened.

**Pull request closed without merging**